bundle/src/main/java/com/adobe/acs/commons/util/OverridePathSlingRequestWrapper.java

```diff
@@ -80,6 +82,22 @@ public OverridePathSlingRequestWrapper(final SlingHttpServletRequest request, fi
         this.myBindings.put(WCMBindings.CURRENT_PAGE, currentPage);
     }
 
+    /**
+     * Gets the getAdapterManager from slingBindings.getSling() or the current bundle context.
+     * Return null if something goes wrong.
+     *
+     * @param slingBindings the slingBindings to get Sling Services
+     * @return        the AdapterManager from slingBindings.getSling() or the current bundle context
+     */
+    public static AdapterManager getAdapterManager(SlingBindings slingBindings) {
+        if (slingBindings != null && slingBindings.getSling() != null) {
+            return slingBindings.getSling().getService(AdapterManager.class);
+        } else {
+            BundleContext bundleContext = FrameworkUtil.getBundle(AdapterManager.class).getBundleContext();
+            return (AdapterManager) bundleContext.getService(bundleContext.getServiceReference(AdapterManager.class.getName()));
```

Comment:
This technically creates a resource leak since the service reference is never "ungot". This also doesn't handle the case when the `AdapterManager` service is unregistered and reregistered. There's two alternatives I can suggest:

1. Move this logic to `com.adobe.acs.commons.util.impl.Activator` and use a `ServiceTracker` to track the service.
2. Deprecate the two-argument constructor and add the BundleContext as a 3rd argument. In this case, the AdapterManager service reference should be got and ungot inside the `adaptTo()` method. In theory, it would be possible to use a `ServletRequestListener` but given that this is a wrapper, it seems like it would be quite complex to discover the wrapper based on the destruction of the "true" request.

Reply:
Thanks for your suggestions.

- to 1.
I guess using Activator could be to much for a service
- to 2.
Do you mean something like this:
```
            if (result == null) {
                AdapterManager mgr = getAdapterManager(myBindings);
                if (mgr == null) {
 ```
instead of:
```
            if (result == null) {
                AdapterManager mgr = this.adapterManager;
                if (mgr == null) {
 ```
Do you know how to test this adaptTo in OverridePathSlingRequestWrapper, it isn't called during htl resolution or adaptTo in Servlet.

Reply:
1. I'm not sure what you mean by "too much". I don't think it is substantially more lines of code.
2. Not exactly (and this is why I'd suggest the first option). What I think you'd have to do is try to get the adapterManager from the bindings and store that in the instance variable. Then, in `adaptTo()` if that instance variable is null, use the service registry to get the AdapterManager, do the adaptation and then unget the service reference.

I believe the easiest way to test this would be to use `registerAdapter` inside the unit test.

I created #2230 as an alternate solution